internal/functions/serve/templates/main.ts

```diff
@@ -0,0 +1,92 @@
+import { serve } from "https://deno.land/std@0.131.0/http/server.ts"
+import * as jose from "https://deno.land/x/jose@v4.13.1/index.ts";
+
+const JWT_SECRET = Deno.env.get("JWT_SECRET");
+const VERIFY_JWT = Deno.env.get("VERIFY_JWT") === "true";
+
+function getAuthToken(req: Request) {
+  const authHeader = req.headers.get("authorization");
+  if (!authHeader) {
+    throw new Error("Missing authorization header");
+  }
+  const [bearer, token] = authHeader.split(" ");
+  if (bearer !== "Bearer") {
+    throw new Error(`Auth header is not 'Bearer {token}'`);
+  }
+  return token;
+}
+
+async function verifyJWT(jwt: string): Promise<boolean> {
+  const encoder = new TextEncoder();
+  const secretKey = encoder.encode(JWT_SECRET);
+  try {
+    await jose.jwtVerify(jwt, secretKey);
+  } catch (err) {
+    console.error(err);
+    return false;
+  }
+  return true;
+}
+
+serve(async (req: Request) => {
+  if (req.method !== "OPTIONS" && VERIFY_JWT) {
+    try {
+      const token = getAuthToken(req);
+      const isValidJWT = await verifyJWT(token);
+
+      if (!isValidJWT) {
+        return new Response(
+          JSON.stringify({ msg: "Invalid JWT"}),
+          { status: 401, headers: { "Content-Type": "application/json" } },
+        ) 
+      }
+    } catch (e) {
+      console.error(e);
+      return new Response(
+        JSON.stringify({ msg: e.toString()}),
+        { status: 401, headers: { "Content-Type": "application/json" } },
+      )
+    }
+  } 
+
+  const url = new URL(req.url);
+  const {pathname} = url;
+  const path_parts = pathname.split("/");
+  const service_name = path_parts[1];
+
+  if (!service_name || service_name === "") {
+    const error = { msg: "missing function name in request" }
+    return new Response(
+        JSON.stringify(error),
+        { status: 400, headers: { "Content-Type": "application/json" } },
+    )
+  }
+
+  const servicePath = `{{ .FuncDir }}/${service_name}`;
```

Comment:
Since we know the `service_name`, shall we add per function verify-jwt option to config.toml?

Reply:
yep, we can do a per function config for both import map and verify jwt. I'm not familiar with parsing toml files, so if you can help with an update, that'd be great.

Reply:
My bad. Seems like we already support per function [VerifyJWT](https://github.com/supabase/cli/blob/main/internal/utils/config.go#L153) so no need to update config toml.

We just need to update main.ts and [ServeFunctions](https://github.com/supabase/cli/blob/main/internal/functions/serve/serve.go#L282) to use the configured value.

Reply:
Perhaps a simple way is to use per function env var such as `VERIFY_JWT_<SERVICE_NAME>`

Reply:
Cool, we can do it as a follow up task. You can provide it as environment variables or as a template variable to the main function.

Reply:
Currently this will introduce yet another different behaviour for `supabase start` and `functions serve`, in addition to the differences with legacy functions serve. I think it's better to complete it in this PR.

Reply:
Is this different from how `serve --all` works currently? Otherwise I can work on supporting this today/tomorrow.

Reply:
On second thought, users can now specify `VERIFY_JWT=false` in `supabase/functions/.env` to support the use case for `supabase start`. So I think it's not really a blocker.